Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -90,165 +83,1086 @@ function Sync-AzSOfflineMarketplaceItem {
+            $productInfo['fileContainers'] = $containerIds
+            $productDetailsProperties['dependentProducts'] = $downloadDetails.properties.dependentProducts
+            $productDetailsProperties['fileContainers'] = $downloadDetails.properties.fileContainers
+        } elseif ($downloadDetails.productKind -eq 'virtualMachine') {
+            $productDetailsProperties['OsDiskImage'] = $downloadDetails.properties.OsDiskImage
+            $productDetailsProperties['DataDiskImages'] = $downloadDetails.properties.DataDiskImages
+        } else {
+            $productDetailsProperties['vmOsType'] = $downloadDetails.properties.vmOsType
+            $productDetailsProperties['sourceBlob'] = $downloadDetails.properties.sourceBlob
+            $productDetailsProperties['computeRole'] = $downloadDetails.properties.computeRole
+            $productDetailsProperties['vmScaleSetEnabled'] = $downloadDetails.properties.vmScaleSetEnabled
+            $productDetailsProperties['supportMultipleExtensions'] = $downloadDetails.properties.supportMultipleExtensions
+            $productDetailsProperties['isSystemExtension'] = $downloadDetails.properties.isSystemExtension
+        }
+
+        $productInfo['links'] = $productDetails.properties.links
+        $productInfo['iconUris'] = $productDetails.properties.iconUris
+        $productInfo['galleryPackageBlobSasUri'] = $downloadDetails.galleryPackageBlobSasUri
+
+        $productDetails.properties|select publisherIdentifier,offer,sku,productKind,vmExtensionType  |out-file "$productFolder\$azpkgName.txt" -Append
+        $productDetails.properties.productProperties|select version| out-file "$productFolder\$azpkgName.txt" -Append
+
+        # select premium download
+        Write-Host $("-"*20)
+        $downloadConfirmation = Read-Host "Downloading package files. Would you like to use Premium download? This requires Azure Storage Tools to be installed. (Y/N)?"
+
+        if ($downloadDetails.productKind -ne 'resourceProvider')
+        {
             # download azpkg
             $azpkgsource = $downloadDetails.galleryPackageBlobSasUri
-            $FileExists = Test-Path "$destination\$azpkgName.azpkg"
-            $DestinationCheck = Test-Path $destination
+            $FileExists = Test-Path "$productFolder\$azpkgName.azpkg"
+            $DestinationCheck = Test-Path $productFolder
             If ($DestinationCheck -eq $false) {
-                new-item -ItemType Directory -force $destination
+                New-item -ItemType Directory -force $productFolder | Out-Null
             }
-            else {}
 
-            If ($FileExists -eq $true) {Remove-Item "$destination\$azpkgName.azpkg" -force} else {
-                New-Item "$destination\$azpkgName.azpkg"
+            If ($FileExists -eq $true) {Remove-Item "$productFolder\$azpkgName.azpkg" -force | Out-Null}
+            $azpkgdestination = "$productFolder\$azpkgName.azpkg"
+
+            If ($downloadConfirmation -eq 'Y') {
```

Comment:
why prompt before checking if the tool is even installed? Will you download and install the tool for them?

Reply:
it might be best to have a switch on your function, something like $EnforcePremiumDownload, and then have the behavior be such that if its installed, you use it, and if it isn't, then you fallback to "normal" download after printing a wanring perhaps, unless the switch was specified in which case you terminate by throwing an exception (perform this check at the start). something to consider.

Reply:
Will consider that. Previously, the script asks more prompts than this, currently don't want to make the tool a requiment.